src/compiler/transformers/module/module.ts

```diff
@@ -545,14 +547,24 @@ namespace ts {
             //          : new Promise(function (_a, _b) { require([x], _a, _b); }); /*Amd Require*/
             // });
             needUMDDynamicImportHelper = true;
-            return createConditional(
-                /*condition*/ createIdentifier("__syncRequire"),
-                /*whenTrue*/ transformImportCallExpressionCommonJS(node),
-                /*whenFalse*/ transformImportCallExpressionAMD(node)
-            );
+            if (isSimpleCopiableExpression(arg)) {
+                return createConditional(
+                    /*condition*/ createIdentifier("__syncRequire"),
+                    /*whenTrue*/ createImportCallExpressionCommonJS(arg),
+                    /*whenFalse*/ createImportCallExpressionAMD(isGeneratedIdentifier(arg) ? arg : getSynthesizedClone(arg)) // TODO (weswigham): Use getSynthesizedDeepClone instead of getSynthesizedClone once #18979 is merged

```

Comment:
I wouldn't use `getDeepSynthesizedClone` for this. Any property-access can have side-effects.

Reply:
Also, while calling `getSynthesizedClone` is correct, when `arg` is a string literal with single quotes, this results in a string literal with double quotes which you can see in the test result below. Generally we try to preserve the quote style, and you can achieve that by calling `setTextRange` on the clone. To avoid duplicating comments you can also call `setNodeEmitFlags(..., NodeEmitFlags.NoComments)`

Reply:
Done